src/plugins/gui/providers/kadoProvider.ts

```diff
@@ -613,10 +616,32 @@ export const kadoProvider: FiatProviderFactory = {
             }
             console.log('Launching Kado webview url=' + url.href)
 
+            // If Kado needs to launch the Plaid bank linking widget, it needs it in an external
+            // webview to prevent some glitchiness. When needed, Kado will send an onMessage
+            // trigger with the url to open. The below code is derived from Kado's sample code
+            const onMessage = (data: string) => {
+              try {
+                datelog(`**** Kado onMessage ${data}`)
+                const message = JSON.parse(data)
+                if (message && message?.type === 'PLAID_NEW_ACH_LINK') {
+                  const achLink = message?.payload?.link
+                  showUi.openExternalWebView({ url: achLink }).catch(e => {
```

Comment:
This archLink could be undefined and not string. How did this not error on the static type checker?

Reply:
not sure. but fixed with cleaner